src/globus_sdk/experimental/tokenstorage/base.py

```diff
@@ -216,11 +225,39 @@ def _default_globus_app_filepath(
         )
 
 
+# https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions
+_RESERVED_FS_CHARS = re.compile(r'[<>:"/\\|?*]')
+# https://stackoverflow.com/a/31976060
+_RESERVED_FS_NAMES = re.compile(r"con|prn|aux|nul|com\d|lpt\d")
+
+
 def _slugify_app_name(app_name: str) -> str:
     """
     Slugify a globus app name for use in a file path.
 
-    This "slugification" will replace spaces with hyphens and lowercase the string so:
-      "My App" -> "my-app"
+    * Reserved filesystem characters are replaced with a '+'. ('a?' -> 'a+')
+    * Periods and Spaces are replaced with a '-'. ('a. b' -> 'a--b')
+    * Control characters are removed. ('a\0b' -> 'ab')
+    * The string is lowercased. ('AB' -> 'ab')
+
+    :raises: GlobusSDKUsageError if the app name is empty after slugification.
+    :raises: GlobusSDKUsageError if the app name is a reserved filesystem name (after
+        slugification).
     """
-    return app_name.replace(" ", "-").lower()
+    app_name = _RESERVED_FS_CHARS.sub("+", app_name)
+    app_name = app_name.replace(".", "-").replace(" ", "-")
+    app_name = "".join(c for c in app_name if c.isprintable())
+    app_name = app_name.lower()
+
+    if _RESERVED_FS_NAMES.fullmatch(app_name):
+        msg = (
+            f'App name results in a reserved filename ("{app_name}"). Please choose a'
+            "different name."
```

Comment:
I recommend breaking at sentences; my experience has been that it's easier to find things when copied-and-pasted error messages aren't broken midway through the line.

```suggestion
            f'App name results in a reserved filename ("{app_name}"). '
            "Please choose a different name."
```

Reply:
¯\_(ツ)_/¯

I don't agree so much as a rule.
In this instance it can't hurt, so I'm happy to change. But while I agree it's an improvement to searching to have a sentence not split over two lines I won't always do that as I think other things, like line management & structure is more important.